cmd/antrea-agent/options.go

```diff
@@ -496,9 +497,54 @@ func (o *Options) validateExternalNodeOptions() error {
 	if unsupported != nil {
 		return fmt.Errorf("unsupported features on Virtual Machine: {%s}", strings.Join(unsupported, ", "))
 	}
+	if err := o.validatePolicyBypassRulesConfig(); err != nil {
+		return fmt.Errorf("policyBypassRules configuration is invalid: %w", err)
+	}
 	return nil
 }
 
+func (o *Options) validatePolicyBypassRulesConfig() error {
+	if len(o.config.PolicyBypassRules) == 0 {
+		return nil
+	}
+	// Each rule contains the following four attributes: direction (in|out), protocol(tcp/udp/icmp/ip),
+	// remote IP or CIDR, dst port (ICMP doesn't require), and these attributes are separated by ":",
+	// e.g. "in:tcp:1.1.1.1:22".
+	for _, rule := range o.config.PolicyBypassRules {
+		attributes := strings.Split(rule, ":")
+		if (len(attributes) < 3) || (len(attributes) > 4) {
+			return fmt.Errorf("the number of attributes: %d for policyBypassRule %s is invalid", len(attributes), rule)
+		}
+		if (attributes[0] != "in") && (attributes[0] != "out") {
+			return fmt.Errorf("direction %s for policyBypassRule %s is invalid", attributes[0], rule)
+		}
+		if (attributes[1] != "tcp") && (attributes[1] != "udp") && (attributes[1] != "icmp") && (attributes[1] != "ip") {
+			return fmt.Errorf("protocol %s for policyBypassRule %s is invalid", attributes[1], rule)
+		}
+		if strings.Contains(attributes[2], "/") {
+			_, _, err := net.ParseCIDR(attributes[2])
+			if err != nil {
+				return fmt.Errorf("cidr %s for policyBypassRule %s is invalid", attributes[2], rule)
+			}
+		} else {
+			ip := net.ParseIP(attributes[2])
+			if ip == nil {
+				return fmt.Errorf("ip %s for policyBypassRule %s is invalid", attributes[2], rule)
+			}
+		}
+		if (len(attributes) == 3) && (attributes[1] != "icmp") {
+			return fmt.Errorf("missing port for policyBypassRule %s", rule)
+		}
+		if len(attributes) == 4 {
+			_, err := strconv.Atoi(attributes[3])
```

Comment:
We should check the port number value falls into the valid range too.

Reply:
Sure, I have added the check to make sure it ranges from 0 to 65535.